vignettes/download_github_comments.Rmd

```diff
@@ -27,40 +36,41 @@ require(magrittr)
+create_file_directory(conf)
 
-owner <- get_github_owner(conf, "project_key_1") # Has to match github organization (e.g. github.com/sailuh)
-repo <- get_github_repo(conf, "project_key_1") # Has to match github repository (e.g. github.com/sailuh/perceive)
 # your file github_token (a text file) contains the GitHub token API
 token <- scan("~/.ssh/github_token",what="character",quiet=TRUE)
 ```
 
-# Issues
+# GitHub Project's Comments
+
+In this notebook, we are interested in obtaining comment data from the GitHub API. Development communication may occur in either issues or pull requests. [The GitHub API Pulls documentation](https://docs.github.com/en/rest/reference/pulls) states that _'Comments on pull requests can be managed via the Issue Comments API. Every pull request is an issue, but not every issue is a pull request. For this reason, "shared" actions for both features, like manipulating assignees, labels and milestones, are provided within the Issues API.'_
+
+Further details are noted on the issue endpoint: _'Note: GitHub's REST API v3 considers every pull request an issue, but not every issue is a pull request. For this reason, "Issues" endpoints may return both issues and pull requests in the response. You can identify pull requests by the pull_request key. Be aware that the id of a pull request returned from "Issues" endpoints will be an issue id. To find out the pull request id, use the "List pull requests" endpoint.'_
+
+While the above is true for **comments**, the first message of every issue and every pull request (which also include a title) is not regarded by GitHub as a comment. For example, suppose one issue was opened by Author A, and contain only one reply by Author B. In this case, A and B established communication, and we would like this interaction to be reflected in the final table. However, if we were to only use the **comments** endpoint, we would only obtain Author's B comment, and not Author's A. The same is true in Pull Requests.
 
+Therefore, in this Notebook we have to rely on four endpoints from the GitHub API: The `Issue endpoint` to obtain the "first comment" of every issue, the `Pull Request endpoint` to obtain the "first comment" of every pull request, followed by the `Issue and Pull Request Comment endpoint`, which provides comments for both issue and pull requests together.
+Unfortunately, neither the `Pull Request endpoint` nor the `Issue and Pull Request Comment endpoint` retrieve in-line code comments from Pull Requests. For these, we need to rely on the fourth endpoint, `Pull Request Comment endpoint`.
+
+# Issues
 ## Issues by Date Range
 
 Kaiaulu offers the ability to download issue data by date range. Using the `github_api_project_issue_by_date()`, you can download issues based on their ['created'](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests#search-by-when-an-issue-or-pull-request-was-created-or-last-updated) dates. The acceptable formats are:
```

Comment:
I think format information would be better in a wiki for date-time formats, though discuss that with Carlos- as I am speaking on what he suggested for me to do.